src/features/admin/routes/item/item-form.tsx

```diff
@@ -0,0 +1,164 @@
+  const formValues = useForm({
+    defaultValues: item,
+  });
+  const { handleSubmit, register, control,  } = formValues;
+
+  const updateItem = (data: ItemData) => {
+    onSubmit(data);
+  };
+
+  const { fields } = useFieldArray({
+    control,
+    name: "availableVariants",
+    keyName: "variantControlId",
+  });
+
+  const { fields: optionFields } = useFieldArray({
+    control,
+    name: "availableOptions",
+    keyName: "optionControlId",
+  });
+
+  return (
+    <FormProvider {...formValues}>
+      <form onSubmit={handleSubmit(updateItem)}>
+        <Stack spacing={2}>
+          <FormControl>
+            <TextField label="Name" {...register("itemName", )}></TextField>
+          </FormControl>
+          <FormControl>
+            <TextField
+              label="Abkürzung"
+              {...register("abbreviation")}
+            ></TextField>
+          </FormControl>
+          <FormControl>
+            <InputLabel id="item-category-label">Kategorie</InputLabel>
+            <Controller
+              name="categoryId"
+              control={control}
+              rules={{ required: true }}
+              render={({ field }) => (
+                <Select
+                  label="Kategorie"
+                  labelId="item-category-label"
+                  {...field}
+                >
+                  {item.availableCategories.map((category) => (
+                    <MenuItem key={category.id} value={category.id}>
+                      {category.name}
+                    </MenuItem>
+                  ))}
+                </Select>
+              )}
+            />
+          </FormControl>
+          <Box>
+            <Typography variant="h4">Varianten</Typography>
+            <Stack mt={2} spacing={4}>
+              {fields.map((variant, index) => (
+                <ItemVariantForm
+                  key={variant.variantControlId}
+                  variant={variant}
+                  index={index}
+                />
+              ))}
+            </Stack>
+          </Box>
+          <Box>
+            <Typography variant="h4">Optionen</Typography>
+            <Stack mt={2}>
+              {optionFields.map((option, index) => (
+                <Box
+                  key={option.optionControlId}
+                  display="flex"
+                  alignItems="center"
+                  justifyContent="space-between"
+                >
+                  <Controller
+                    name={`availableOptions.${index}.selected`}
+                    control={control}
+                    render={({ field }) => (
+                      <Checkbox checked={field.value} {...field} />
+                    )}
+                  />
+                  <Typography>{option.name}</Typography>
+                  <TransformController
+                    name={`availableOptions.${index}.defaultValue`}
+                    control={control}
+                    transform={{
```

Comment:
Can we extract this `transform` object to a constant? Maybe in the same manner we have for the other transformers?

You can also add a TODO to not lose track.

Reply:
added todo